projects/gsc/experiments/base.py

```diff
@@ -0,0 +1,181 @@
+#  Numenta Platform for Intelligent Computing (NuPIC)
+#  Copyright (C) 2020, Numenta, Inc.  Unless you have an agreement
+#  with Numenta, Inc., for a separate license for this software code, the
+#  following terms and conditions apply:
+#
+#  This program is free software you can redistribute it and/or modify
+#  it under the terms of the GNU Affero Public License version 3 as
+#  published by the Free Software Foundation.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#  See the GNU Affero Public License for more details.
+#
+#  You should have received a copy of the GNU Affero Public License
+#  along with this program.  If not, see htt"://www.gnu.org/licenses.
+#
+#  http://numenta.org/licenses/
+#
+
+"""
+Base Imagenet Experiment configuration.
+"""
+
+import os
+import sys
+from copy import deepcopy
+
+import torch
+
+from nupic.research.frameworks.pytorch.datasets import preprocessed_gsc
+from nupic.research.frameworks.pytorch.imagenet import RezeroedKWinnersGSCExperiment
+from nupic.research.frameworks.pytorch.models.le_sparse_net import LeSparseNet
+from nupic.torch.models.sparse_cnn import gsc_sparse_cnn
+
+# Batch size depends on the GPU memory.
+BATCH_SIZE = 16
+
+# Default configuration based on Pytorch GSC training example.
+# This is a replicates the config of `sparseCNN2` from
+#   `nupic.research\projects\whydense\gsc\experiments_v3.cfg`
+# This achieves 94.7073% accuracy
+DEFAULT = dict(
+    experiment_class=RezeroedKWinnersGSCExperiment,
+
+    # Results path
+    local_dir=os.path.expanduser("~/nta/results/experiments/gsc"),
+
+    # Dataset
+    dataset_class=preprocessed_gsc,
+    dataset_args=dict(
+        data_dir="~/nta/datasets/gsc_preprocessed",
+        download=True,
+    ),
+
+    # Seed
+    seed=20,
+
+    # Number of times to sample from the hyperparameter space. If `grid_search` is
+    # provided the grid will be repeated `num_samples` of times.
+    num_samples=1,
+
+    # Training batch size
+    batch_size=BATCH_SIZE,
+    # Validation batch size
+    val_batch_size=BATCH_SIZE,
+    # Number of batches per epoch. Useful for debugging
+    batches_in_epoch=5121,
```

Comment:
I would use `sys.maxsize` here just in case someone updates the `BATCH_SIZE`.
See https://github.com/NumentaCorp/nupic.hardware/blob/master/projects/imagenet/experiments/base.py#L56